Classes/Backend/TceMain.php

```diff
@@ -102,11 +187,18 @@ public function processDatamap_postProcessFieldArray($status, $table, $id, &$fie
      */
     public function processCmdmap_postProcess(&$command, $table, $id, &$relativeTo, &$reference, &$pasteUpdate, &$pasteDataMap)
     {
-        if ($table === 'tt_content' && ($command === 'localize' || $command === 'copy' || $command === 'copyToLanguage')) {
+        $doProcessCommand = in_array($command,['localize','copy','copyToLanguage','delete']);
+        $doProcessCommand = $doProcessCommand && 'tt_content' == $table;
+
+        if ($doProcessCommand) {
             // A copy, or localisation (which is also a copy) was made. Cascade copy operations for child records.
-            $recordsToCopy = [];
+            $gridChildRecords = [];
             $resolver = $this->objectManager->get(ProviderResolver::class);
             $originalRecord = $this->recordService->getSingle($table, '*', $id);
+            $recordCanBeGrid = !empty($originalRecord['pi_flexform']);
```

Comment:
Looks like we have significant duplication here - maybe there's a way we can refactor!